### PR TITLE
chore(MeshTrafficPermission): use http rbac envoy filter when applicable

### PR DESCRIPTION
**Pull request by bartsmykla**
Instead of always using `envoy.filters.network.rbac` we are gonna use `envoy.filters.http.rbac` when applicable.

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues
  - Closes https://github.com/kumahq/kuma/issues/5045
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS
  - It won't
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s)
  - Unit tests were updated
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)?
  - There is no need
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label)
  - There is no need

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/testdata/apply-egress.golden.yaml [hunk 1]
```diff
@@ -13,9 +13,9 @@ resources:
         - external-service-1{mesh=mesh-1}
         transportProtocol: tls
       filters:
-      - name: envoy.filters.network.rbac
+      - name: envoy.filters.http.rbac
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/testdata/apply-egress.golden.yaml:16]: have you checked that it actually works? I thought that HTTP filters have to be placed in httpFilters inside of HTTP connection manager

Comment (bartsmykla) [reply pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/testdata/apply-egress.golden.yaml:16]: Good catch.

Comment (bartsmykla) [reply pkg/plugins/policies/meshtrafficpermission/plugin/v1alpha1/testdata/apply-egress.golden.yaml]: addressed

**Review by jakubdyszkiewicz — COMMENTED** [review]
We use TCP RBAC stats in Grafana dashboards. Can you check what happens there if we use HTTP RBAC filter?

File: test/e2e_env/universal/meshtrafficpermission/meshtrafficpermission.go [hunk 2]
```diff
@@ -57,7 +57,7 @@ func MeshTrafficPermissionUniversal() {
 				universal.Cluster, "demo-client", "test-server.mesh",
 			)
 			g.Expect(err).ToNot(HaveOccurred())
-			g.Expect(response.ResponseCode).To(Equal(503))
+			g.Expect(response.ResponseCode).To(Equal(403))
```
Comment (lukidzi) [inline test/e2e_env/universal/meshtrafficpermission/meshtrafficpermission.go:60]: Could we have an another case for TCP traffic?

**Comment by bartsmykla** [comment]
> We use TCP RBAC stats in Grafana dashboards. Can you check what happens there if we use HTTP RBAC filter?

I've checked on local k3d cluster and it seems that stats `envoy_rbac_*` stats we are using in dashboards are still correctly updated

File: test/e2e_env/universal/meshtrafficpermission/meshtrafficpermission.go (hunk 2 above)
Comment (bartsmykla) [reply test/e2e_env/universal/meshtrafficpermission/meshtrafficpermission.go]: added

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**